source/MaterialXGenOgsXml/OgsFragment.h

```diff
@@ -44,7 +45,7 @@ class OgsFragment
     /// Return the MaterialX document.
     mx::DocumentPtr getDocument() const
     {
-        return _document;
+        return _element ? _element->getDocument() : mx::DocumentPtr();
```

Comment:
Not storing the document anymore because it's referenced by the element. This helps simplify the design which was necessary for testability.